docs/tech-notes/observability/index_recommendations.md

```diff
@@ -0,0 +1,79 @@
+# Index recommendations on SQL Statistics
+Last Update: February 2024
+Original author: maryliag
+
+This document provides an overview of how Index Recommendations are 
+generated and stored on statement statistics.
+
+Table of contents:
+
+- [Overview](#overview)
+- [Cache](#cache)
+- [Cluster Settings](#cluster-settings)
+- [Console](#console)
+
+## Overview
+When a statement is executed, we want to calculate index recommendations for it, in
+case there are better indexes that can improve its execution. Generating an index
+recommendation is a costly operation, so we don't want to generate it on each execution. 
+At the same time, generating frequent index recommendations for the same plan would
+likely generate the same results (unless there is a big increase/decrease in rows affected
+by the query).
+With this in mind, we use a cache to store the latest index recommendation for the most
+recent statement fingerprints, avoiding a new generation on each execution.
+
+The value of the recommendation can be found on the column `index_recommendation`
+as a `STRING[] NOT NULL` on:
+- `system.statement_statistics`
+- `crdb_internal.statement_statistics`
+- `crdb_internal_statement_statistics_persisted`
+
+When recording a statement, it calls the function `ShouldGenerateIndexRecommendation`.
+This function returns true if there was no index recommendation generated for the
+statement fingerprint in the past hour and there was at least 5 executions 
+(`minExecCount`) of it.
+We don't generate index recommendations for statement fingerprints with 5 or fewer 
+executions because we don't want to perform a heavy operation for a statement that
```

Comment:
How heavy of an operation is generating index recommendations?

Reply:
I added a footnote on the cost of generation and what factors in it